ui/app/AppLayouts/Wallet/controls/SortOrderComboBox.qml

```diff
@@ -0,0 +1,272 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtGraphicalEffects 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Popups 0.1
+
+import utils 1.0
+
+ComboBox {
+    id: root
+
+    property int sortOrder: Qt.DescendingOrder
+    readonly property string currentSortRoleName: d.currentSortRoleName
+
+    model: d.predefinedSortModel
+    textRole: "text"
+    valueRole: "value"
+    displayText: !d.isCustomSortOrder ? "%1 %2".arg(currentText).arg(sortOrder === Qt.DescendingOrder ? "↓" : "↑")
+                                      : currentText
+
+    Component.onCompleted: currentIndex = indexOfValue(SortOrderComboBox.TokenOrderCustom)
+
+    enum TokenOrder {
+        TokenOrderNone = 0,
+        TokenOrderCustom,
+        TokenOrderValue,
+        TokenOrderBalance,
+        TokenOrder1WChange,
+        TokenOrderAlpha
+    }
+
+    horizontalPadding: 12
+    verticalPadding: 8
+    spacing: 8
+
+    font.family: Theme.palette.baseFont.name
+    font.pixelSize: Style.current.additionalTextSize
+
+
+    QtObject {
+        id: d
+
+        readonly property int defaultDelegateHeight: 34
+
+//        // models
+//        readonly property SortFilterProxyModel tokensModel: SortFilterProxyModel {
+//            sourceModel: root.baseModel
+//            proxyRoles: [
+//                ExpressionRole {
+//                    name: "currentBalance"
+//                    expression: model.enabledNetworkBalance.amount
+//                },
+//                ExpressionRole {
+//                    name: "currentCurrencyBalance"
+//                    expression: model.enabledNetworkCurrencyBalance.amount
+//                }
+//            ]
+//            sorters: RoleSorter {
+//                roleName: cmbTokenOrder.currentSortRoleName
+//                sortOrder: cmbTokenOrder.sortOrder
+//                enabled: !d.isCustomSortOrder
+//            }
+//            filters: ValueFilter {
+//                roleName: "visibleForNetworkWithPositiveBalance"
+//                value: true
+//            }
+//        }
```

Comment:
leftovers?

Reply:
Yes and no, this will be needed but as I said, this whole component can be ignored for now (Khushboo and I will need it at some point)